src/shared/parse.cpp

```diff
@@ -1123,6 +1123,12 @@ Parse_StripDoubleQuotes
 */
 static void Parse_StripDoubleQuotes( char *string )
 {
+	if ( string == nullptr || *string == '\0' )
+	{
+		Log::Warn( "Parse_StripDoubleQuotes called with invalid or empty string" );
+		return;
+	}
+
 	if ( *string == '\"' )
 	{
 		memmove( string, string + 1, strlen( string ) + 1 );
```

Comment:
Are you sure the UB is coming from the empty string case? Because it can also read out of bounds if the string begins with a quote. This line is wrong; it should be `memmove( string, string + 1, strlen( string ) );`

Reply:
The actual problem UBsan found was not from the memmove, but from the `string[ strlen( string ) - 1 ] = '\0';` line. 

The path I propose mostly prevent it from silently corrupting memory, instead it prevents said corruption and emits noise when such situation would happen.
I have not dug further since I was debugging a different, unrelated issue of my branch, but since this one could easily be ported to master, I thought it would be a good idea.
You are, indeed, right about the memmove too, let me fix it in that PR.